forward mouse events from textareas to canvas

fahdarafat opened the pull request:
This should fix [1773](https://github.com/Comfy-Org/ComfyUI_frontend/issues/1773).
I'm not sure though if there are more DOM elements to handle. This currently only handles textareas




https://github.com/user-attachments/assets/3811eddd-9d01-4c8c-a918-a801e6d0db01

┆Issue is synchronized with this [Notion page](https://www.notion.so/1911-forward-mouse-events-from-textareas-to-canvas-15c6d73d3650810f8c4cdc68d026c862) by [Unito](https://www.unito.io)

fahdarafat committed:
forward mouse events from textareas to canvas

fahdarafat committed:
use arrow functions

huchenlei commented:
@webfiltered Can you take a look at this PR?

webfiltered reviewed (changes requested):
Awesome, thank you!  I had this on my to-do list, so it's much appreciated!

(this review looks like a lot of changes, but really it's just three very minor ones)

webfiltered committed:
Apply suggestions from code review

Use pointer events instead of mouse to get extended properties
Remove workaround (only required due to missing extended properties)

webfiltered committed:
Apply code review - Adjust UX

- Do not prevent middle click from hitting DOM widgets
- Allow middle click and drag to continue functioning if another button is held down
